polaris-react/src/components/Button/Button.scss

```diff
@@ -607,6 +625,11 @@
       --pc-button-color-active: var(
         --p-color-bg-transparent-active-experimental
       );
+
+      &:hover,
+      &:active {
+        --pc-button-text: var(--p-color-text-critical);
+      }
```

Comment:
Unrelated to your PR I pushed up this bug fix I noticed while tophatting:

**Before:** 

https://github.com/Shopify/polaris/assets/32409546/24b1d401-466a-4f03-ade0-fe3263dab88d

**After:**

https://github.com/Shopify/polaris/assets/32409546/d4a123d3-c24d-4c47-9260-d6bec31e7334